# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserChild.cpp
# Commit: 405100db6c45
# Full Hash: 405100db6c45f0463286067df5b491f74790d9aa
# Author: Kashav Madan <kmadan@mozilla.com>
# Date: 2019-07-11 09:51:12
# Regressor Bug: 1543251
# File Overlap Count: 1
# Description:
#   Bug 1543251 - Move hasBeforeUnload from PBrowser to PWindowGlobal, r=nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D37003
# ==============================================================================

diff -r 0c408343080d -r 405100db6c45 dom/ipc/BrowserChild.cpp
--- a/dom/ipc/BrowserChild.cpp	Thu Jul 04 07:41:57 2019 +0000
+++ b/dom/ipc/BrowserChild.cpp	Wed Jul 10 21:13:44 2019 +0000
@@ -355,7 +355,6 @@
       mChromeFlags(aChromeFlags),
       mMaxTouchPoints(0),
       mLayersId{0},
-      mBeforeUnloadListeners(0),
       mEffectsInfo{EffectsInfo::FullyHidden()},
       mDidFakeShow(false),
       mNotified(false),
@@ -3430,26 +3429,6 @@
   return NS_OK;
 }
 
-void BrowserChild::BeforeUnloadAdded() {
-  // Don't bother notifying the parent if we don't have an IPC link open.
-  if (mBeforeUnloadListeners == 0 && IPCOpen()) {
-    SendSetHasBeforeUnload(true);
-  }
-
-  mBeforeUnloadListeners++;
-  MOZ_ASSERT(mBeforeUnloadListeners >= 0);
-}
-
-void BrowserChild::BeforeUnloadRemoved() {
-  mBeforeUnloadListeners--;
-  MOZ_ASSERT(mBeforeUnloadListeners >= 0);
-
-  // Don't bother notifying the parent if we don't have an IPC link open.
-  if (mBeforeUnloadListeners == 0 && IPCOpen()) {
-    SendSetHasBeforeUnload(false);
-  }
-}
-
 NS_IMETHODIMP BrowserChild::BeginSendingWebProgressEventsToParent() {
   mShouldSendWebProgressEventsToParent = true;
   return NS_OK;